webapp/packages/core-data-context/src/DataContext/TempDataContext.ts

```diff
@@ -28,6 +28,7 @@ export class TempDataContext implements IDataContext {
     this.fallback = fallback;
 
     makeObservable<this>(this, {
+      map: observable.shallow,
```

Comment:
should we do same thing for the? to avoid similar bug cases
`webapp/packages/core-data-context/src/DataContext/DynamicDataContext.ts`